nucypher/characters/lawful.py

```diff
@@ -1251,21 +1252,42 @@ def interface_info_with_metadata(self):
         return constants.BYTESTRING_IS_URSULA_IFACE_INFO + bytes(self)
 
     #
-    # Utilities
+    # Work Orders & Re-Encryption
     #
 
-    def work_orders(self, bob=None):
-        """
-        TODO: This is better written as a model method for Ursula's datastore.
-        """
-        if not bob:
-            return self._work_orders
-        else:
-            work_orders_from_bob = []
-            for work_order in self._work_orders:
-                if work_order.bob == bob:
-                    work_orders_from_bob.append(work_order)
-            return work_orders_from_bob
+    def work_orders(self, bob=None) -> List['WorkOrder']:
+        with ThreadedSession(self.datastore.engine):
+            if not bob:  # All
+                return self.datastore.get_workorders()
+            else:        # Filter
+                work_orders_from_bob = self.datastore.get_workorders(bob_verifying_key=bytes(bob.stamp))
+                return work_orders_from_bob
+
+    def _reencrypt(self, kfrag: KFrag, work_order: 'WorkOrder', alice_verifying_key: UmbralPublicKey):
```

Comment:
I felt that re-encrypt is not a public method like the others since it isn't meant to be called manually.  I can still change it if thats the general sentiment.